Spigot-Server-Patches/0700-add-get-set-biome-for-namespaced-keys.patch

```diff
@@ -0,0 +1,150 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Thonk <30448663+ExcessiveAmountsOfZombies@users.noreply.github.com>
+Date: Sun, 11 Apr 2021 21:43:48 -0500
+Subject: [PATCH] add get set biome for namespaced keys
+
+
+diff --git a/src/main/java/net/minecraft/core/IRegistry.java b/src/main/java/net/minecraft/core/IRegistry.java
+index 4f04d8081912e0fe771f0db9e086c789328f246f..4bcf5971cf57c5d75ca876ddc57197095785f39e 100644
+--- a/src/main/java/net/minecraft/core/IRegistry.java
++++ b/src/main/java/net/minecraft/core/IRegistry.java
+@@ -229,7 +229,7 @@ public abstract class IRegistry<T> implements Codec<T>, Keyable, Registry<T> {
+     public static final ResourceKey<IRegistry<StructureFeature<?, ?>>> av = a("worldgen/configured_structure_feature");
+     public static final ResourceKey<IRegistry<ProcessorList>> aw = a("worldgen/processor_list");
+     public static final ResourceKey<IRegistry<WorldGenFeatureDefinedStructurePoolTemplate>> ax = a("worldgen/template_pool");
+-    public static final ResourceKey<IRegistry<BiomeBase>> ay = a("worldgen/biome");
++    public static final ResourceKey<IRegistry<BiomeBase>> ay = a("worldgen/biome"); public static ResourceKey<IRegistry<BiomeBase>> getBiomeKeys() { return ay; } // paper - OBFhelper
+     public static final ResourceKey<IRegistry<WorldGenSurface<?>>> az = a("worldgen/surface_builder");
+     public static final IRegistry<WorldGenSurface<?>> SURFACE_BUILDER = a(IRegistry.az, () -> {
+         return WorldGenSurface.v;
+diff --git a/src/main/java/net/minecraft/core/IRegistryCustom.java b/src/main/java/net/minecraft/core/IRegistryCustom.java
+index 489737d8b108bbca4124495faf92cb27f666a750..b32a3221f97c8ccec748c965d0189d8da592d0ac 100644
+--- a/src/main/java/net/minecraft/core/IRegistryCustom.java
++++ b/src/main/java/net/minecraft/core/IRegistryCustom.java
+@@ -70,6 +70,7 @@ public abstract class IRegistryCustom {
+             return new IllegalStateException("Missing registry: " + resourcekey);
+         });
+     }
++    public <E> IRegistryWritable<E> getRegistry(ResourceKey<? extends IRegistry<E>> resourcekey) { return b(resourcekey); } // Paper - OBFHELPER
```

Comment:
OBFHELPERs go above the method they're hooking, wanna make sure stuff conflicts if the header changes

Reply:
I'm editing the PR again, would you like me to put it above and leave a newline above the method or put it directly above the method?

Reply:
directly above, no newlines

Reply:
ok, I moved it above